src/osd/osd_types.cc

```diff
@@ -2956,14 +2958,19 @@ void pg_stat_t::encode(ceph::buffer::list &bl) const
   encode(manifest_stats_invalid, bl);
   encode(avail_no_missing, bl);
   encode(object_location_counts, bl);
-  ENCODE_FINISH(bl);
```

Comment:
this will always be true - you should be able to encode unconditionally and not need new_compat_v or ENCODE_FINISH_NEW_COMPAT.

Old daemons decoding will have extra data that they don't decode at the end, which won't cause any problems.

Reply:
Okay got it. I'll fix this.